### PR TITLE
Update 'Distance from here' case list xpath expressions

### PR DESCRIPTION
**Pull request by philomates**
- For sorting by distance, if there is no gps value for the case property in question, use the max int value. This will make all cases with blank gps values sort to the bottom.
- Show 1 decimal point of precision to get more fine-grained info on values below 1 km.

follow up to work done in https://github.com/dimagi/commcare-hq/pull/10023 and https://github.com/dimagi/commcare-hq/pull/10317

I manually configured a ccz with these changes, but didn't test building an app using a local instance of HQ. 

@snopoke can you take a look at this?

### CODE REVIEW

File: corehq/apps/app_manager/detail_screen.py [hunk 1]
```diff
@@ -279,8 +279,8 @@ class TimeAgo(FormattedDetailColumn):
 
 @register_format_type('distance')
 class Distance(FormattedDetailColumn):
-    XPATH_FUNCTION = u"if(here() = '', '', if({xpath} = '', '', concat(round(distance({xpath}, here()) div 1000), ' km')))"
-    SORT_XPATH_FUNCTION = u'round(distance({xpath}, here()))'
+    XPATH_FUNCTION = u"if(here() = '' or {xpath} = '', '', concat(round(distance({xpath}, here()) div 100) div 10, ' km'))"
```
Comment (dimagimon) [inline corehq/apps/app_manager/detail_screen.py]: E501 line too long (123 > 115 characters)

File: corehq/apps/app_manager/tests/test_case_detail_distance.py [hunk 2]
```diff
@@ -33,12 +33,12 @@ def test_short_detail_xml(self):
...
                     <template>
                         <text>
-                            <xpath function="if(here() = '', '', if(case_name = '', '', concat(round(distance(case_name, here()) div 1000), ' km')))"/>
+                            <xpath function="if(here() = '' or case_name = '', '', concat(round(distance(case_name, here()) div 100) div 10, ' km'))"/>
                         </text>
                     </template>
                     <sort direction="ascending" order="1" type="double">
                         <text>
-                                <xpath function="round(distance(case_name, here()))"/>
+                                <xpath function="if(case_name = '', 2147483647, round(distance(case_name, here())))"/>
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_detail_distance.py]: E501 line too long (118 > 115 characters)

**Comment by snopoke** [comment]
Build failure:
```
======================================================================
FAIL: test_short_detail_xml (corehq.apps.app_manager.tests.test_case_detail_distance.CaseDetailDistance)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_case_detail_distance.py", line 48, in test_short_detail_xml
    template_xpath
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 26, in assertXmlPartialEqual
    self.assertXmlEqual(expected, actual, normalize=False)
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/util.py", line 126, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -12,7 +12,7 @@
     </template>
     <sort direction="ascending" order="1" type="double">
       <text>
-        <xpath function="if(case_name = '', 2147483647, round(distance(case_name, here())))"/>
+        <xpath function="if(case_name = , 2147483647, round(distance(case_name, here())))"/>
       </text>
     </sort>
   </field>
```

File: corehq/apps/app_manager/tests/test_case_detail_distance.py [hunk 3]
```diff
@@ -159,7 +159,7 @@ def test_long_detail_xml(self):
                     </header>
                     <template>
                         <text>
-                            <xpath function="if(here() = '', '', if(case_name = '', '', concat(round(distance(case_name, here()) div 1000), ' km')))"/>
+                            <xpath function="if(here() = '' or case_name = '', '', concat(round(distance(case_name, here()) div 100) div 10, ' km'))"/>
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_detail_distance.py]: E501 line too long (151 > 115 characters)

Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_detail_distance.py]: E501 line too long (151 > 115 characters)

**Pull request merged**